src/meltano/core/hub/client.py

```diff
@@ -198,6 +199,48 @@ def plugin_endpoint(
 
         return url
 
+    def _build_request(self, method: str, url: str) -> requests.PreparedRequest:
+        """Build a request to the Hub API.
+
+        Args:
+            method: The HTTP method.
+            url: The URL to request.
+
+        Returns:
+            The prepared request.
+        """
+        request = requests.Request(method, url)
+
+        try:
+            click_context = click.get_current_context(silent=True)
+        except RuntimeError:
+            logger.debug("Could not get current click context")
+            click_context = None
+
+        if click_context:
+            request.headers.update(
+                {"X-Meltano-Command": click_context.command_path},
+            )
```

Comment:
This is probably the best way to do this currently, but it'll cause some pain if we ever decide to separate the CLI from the core.

Reply:
Yeah, I'm not a fan of calling `click` here but decoupling might require a not-so-small refactor to the `ProjectPluginsService` to make it aware of the request context.

Reply:
(or having a singleton Meltano context in which the CLI sets some attributes)